### PR TITLE
Fix discard message on outbox and improve the discard new message

### PR DESCRIPTION
**Pull request by GretaD**
Fixes https://github.com/nextcloud/mail/issues/6268

just like #6302 but without a failed rebase

### CODE REVIEW

File: src/components/NewMessageModal.vue [hunk 1]
```diff
@@ -187,6 +188,25 @@ export default {
...
+			const isOutbox = this.composerMessage.type === 'outbox'
+			if (isOutbox) {
+				id = this.composerMessage.data.id
+			}
+			this.$emit('close')
+			try {
+				if (isOutbox) {
+					await this.$store.dispatch('outbox/deleteMessage', { id })
+				} else {
+					await this.$store.dispatch('deleteMessage', { id })
```
Comment (ChristophWurst) [inline src/components/NewMessageModal.vue:202]: id is always undefind

Comment (GretaD) [reply src/components/NewMessageModal.vue:202]: Im not sure what am i supposed to do

File: src/components/Composer.vue [hunk 2]
```diff
@@ -1041,16 +1037,8 @@ export default {
...
-			try {
-				id = await this.draftsPromise
-				await this.$store.dispatch('deleteMessage', { id })
-			} catch (err) {
-				logger.error('Could not delete message with id ' + id)
-			}
-			this.state = STATES.DISCARDED
-			this.$emit('close')
+			const id = await this.draftsPromise
+			this.$emit('discard-draft', id)
```
Comment (ChristophWurst) [inline src/components/Composer.vue:1041]: here you emit an id

File: src/components/NewMessageModal.vue [hunk 3]
```diff
@@ -187,6 +188,25 @@ export default {
 				return `"${recipient.label}" <${recipient.email}>`
 			}
 		},
+		async discardDraft() {
```
Comment (ChristophWurst) [inline src/components/NewMessageModal.vue:191]: but you don't use the argument value

**Review by ChristophWurst — APPROVED** [review]
Works

**Pull request merged**